include/misc/mempool.h

```diff
@@ -20,7 +20,7 @@ struct sys_mem_pool_block {
 	struct sys_mem_pool *pool;
 	u32_t level : 4;
 	u32_t block : 28;
-};
+} __aligned(sizeof(void *));
```

Comment:
And while this struct does contain pointers and does need to be pointer-aligned, it's stored inline in the heap block itself.  Putting an alignment requirement here won't actually fix anything, because the compiler never puts these on a stack or in static data.

The (much nastier!) code that locates and aligns the blocks is in lib/os/mempool.c with some complicated macros that compute static sizes based on those rules in mempool_base.h.